src/pages/Dictionary/components/DisplayMode.jsx

```diff
@@ -1,24 +1,124 @@
 import { Button } from '@nextui-org/react';
 import { LikeFilled, DislikeFilled } from '@ant-design/icons';
+import {
+  incrementLikes,
+  incrementDislikes,
+  decrementDislikes,
+  decrementLikes
+} from '../../../utils/firebaseOperations';
+import { useState } from 'react';
 
 const DisplayMode = ({ entry }) => {
+  const [likes, setLikes] = useState(entry.likes);
+  const [dislikes, setDislikes] = useState(entry.dislikes);
+  const [userAction, setUserAction] = useState(null); // 'liked', 'disliked', or null
+
+  const handleLike = async () => {
```

Comment:
this function will be called when likes button is clicked when why do we need to check userAction?

Reply:
oh is it mean unclick the likes\?